app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala

```diff
@@ -227,13 +265,38 @@ sealed trait State {
         Xor.left(FileWithError)
       }
     }).getOrElse(fileNotFoundError)
+  }
+
 }
 
 object NotCreated extends State {
   import State._
 
-  override def canCreate(): CanResult =
-    successResult
+  override def canCreateWithFilesCapacityAndSize(maxFiles: Int, maxSize: String): CanResult =
+    (maxFiles, maxSize)match {
+      case (num, size) => if (num > Envelope.defaultMaxNumFilesCapacity) envelopeMaxNumFilesExceededError
+                          else if (!isValidEnvelopeSize(size)) envelopeMaxSizeExceededError
+                          else successResult
+      case _ => successResult
+    }
+
+  def isValidEnvelopeSize(size: String): Boolean = {
+    val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
+    size.toUpperCase match {
```

Comment:
Agree with @kings13y, it would be good not to duplicate the regex. Maybe put in a companion object.